ultralytics/engine/model.py

```diff
@@ -94,7 +94,13 @@ def __init__(self, model: Union[str, Path] = 'yolov8n.pt', task=None) -> None:
             self._load(model, task)
 
     def __call__(self, source=None, stream=False, **kwargs):
-        """Calls the 'predict' function with given arguments to perform object detection."""
+        """Calls the predict() method with given arguments to perform object detection."""
+        return self.predict(source, stream, **kwargs)
+
+    def embed(self, source=None, stream=False, **kwargs):
+        """Calls the predict() method and returns concatenated embeddings."""
+        if not kwargs.get('embed'):
```

Comment:
@glenn-jocher  We should avoid using kwargs and stick to explicit args to make things easier for new users to understand. IDEs suggest the explicit args with their docstring, but not kwargs..Kwargs should probably be exceptions not rules

Reply:
And probably also add tests. Example
```
def test_embed():
    model = YOLO(MODEL)
    seg_model = YOLO(WEIGHTS_DIR / 'yolov8n-seg.pt')

    batch = [
        str(SOURCE),  # filename
        Path(SOURCE),  # Path
        'https://ultralytics.com/images/zidane.jpg' if ONLINE else SOURCE,  # URI
        cv2.imread(str(SOURCE)),  # OpenCV
        Image.open(SOURCE),  # PIL
        np.zeros((320, 640, 3))]  # numpy

    results = model.embed(batch, imgsz=32)
    results_seg = seg_model.embed(batch, imgsz=32)
    assert len(results) == len(results_seg) == len(batch)  # multiple sources in a batch
    assert results[0].embeddings.shape == results_seg[0].embeddings.shape

```

Reply:
@AyushExel yes I agree. The problem is embed() and predict(), and most of the model methods accept very many arguments, so updating these docstrings is a great idea, but an enormous task that's probably separate from this PR. We ideally want to add all of these arguments:

https://docs.ultralytics.com/modes/predict/#inference-arguments
<img width="958" alt="Screenshot 2023-12-21 at 17 04 30" src="https://github.com/ultralytics/ultralytics/assets/26833433/47a1de3a-b21f-432d-b14a-a19221283d6b">

Reply:
Yeah existing args are fine, we can just avoid adding to it so we don't make it worse